integration-tests/api/__tests__/admin/draft-order/ff-tip-draft-order.js

```diff
@@ -0,0 +1,106 @@
+const path = require("path")
+
+const draftOrderSeeder = require("../../../helpers/draft-order-seeder")
+const adminSeeder = require("../../../helpers/admin-seeder")
+const {
+  simpleDiscountFactory,
+  simpleRegionFactory,
+  simpleShippingOptionFactory,
+} = require("../../../factories")
+const startServerWithEnvironment =
+  require("../../../../helpers/start-server-with-environment").default
+const { useDb } = require("../../../../helpers/use-db")
+const { useApi } = require("../../../../helpers/use-api")
+
+jest.setTimeout(30000)
+
+describe("[MEDUSA_FF_TAX_INCLUSIVE_PRICING] /admin/draft-orders", () => {
+  let medusaProcess
+  let dbConnection
+
+  beforeAll(async () => {
+    const cwd = path.resolve(path.join(__dirname, "..", "..", ".."))
+    const [process, connection] = await startServerWithEnvironment({
+      cwd,
+      env: { MEDUSA_FF_TAX_INCLUSIVE_PRICING: true },
+      verbose: true,
+    })
+
+    dbConnection = connection
+    medusaProcess = process
+  })
+
+  afterAll(async () => {
+    const db = useDb()
+    await db.shutdown()
+
+    medusaProcess.kill()
+  })
+
+  describe("POST /admin/draft-orders", () => {
+    beforeEach(async () => {
+      await adminSeeder(dbConnection)
+      await draftOrderSeeder(dbConnection)
+    })
+
+    afterEach(async () => {
+      const db = useDb()
+      await db.teardown()
+    })
+
+    it("creates a draft order cart", async () => {
+      const api = useApi()
+
+      await simpleRegionFactory(dbConnection, {
+        id: "tip-region",
+        includes_tax: true,
+        currency_code: "usd",
+      })
+
+      await dbConnection.manager.query(
+        `update currency set includes_tax = true where code = 'usd'`
+      )
+
+      await simpleDiscountFactory(dbConnection, {
+        code: "testytest",
+        regions: ["tip-region"],
+      })
+      await simpleShippingOptionFactory(dbConnection, {
+        id: "tip-option",
+        price: 100,
+        region_id: "tip-region",
+      })
+
+      const payload = {
+        email: "oli@test.dk",
+        shipping_address: "oli-shipping",
+        discounts: [{ code: "testytest" }],
+        items: [
+          {
+            variant_id: "test-variant",
+            quantity: 2,
+            metadata: {},
+          },
+        ],
+        region_id: "tip-region",
+        customer_id: "oli-test",
+        shipping_methods: [
+          {
+            option_id: "tip-option",
+          },
+        ],
+      }
+
+      const response = await api
+        .post("/admin/draft-orders", payload, {
+          headers: {
+            Authorization: "Bearer test_token",
+          },
+        })
+        .catch((err) => {
+          console.log(err)
+        })
+      expect(response.status).toEqual(200)
```

Comment:
suggestion: Should we check the discount exists

Reply:
I don't think it is necessary to re-validate in the tax-inclusive context. Discount are already covered by then main draft-order test suite.

Reply:
yeah but we also want to be sure that adding the tax inclusive flag does not change the behaviour and break anything wdyt?

Reply:
Fair point. I mainly concerned about our testing strategy. Should we have integration tests for the whole system in a tax-inclusive context? The answer might be yes but that's a whole undertaking.

Reply:
The thing is that the choice was to completely separate the tests under flag and the other tests. The idea is that the flags are temporary and not meant to be there forever and therefore the tests will be then cleaned up and merged together. In the mean time to ensure that under a flag it does not break any feature, the place where the flag is used need to be re tested the same way it is when there is no flag to unsure that the behaviour is the same including the values which can differ from the test not under the flag. Is that understandable? But no need to migrate all the test under the flag as well, only the places impacted, here the creation with discount.

Reply:
Ok, makes sense now. I'll look at what areas of draft-order are affected by tax-inclusive and add tests for them. I'll have to ask for a re-review of the PR then. Thanks.